### PR TITLE
tiproxy: add docs for label-based balance

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? (Required)

Label-based balance was released in v1.1.0, but it was not described in the docs because we may need to design it carefully but there were no users at that time, so we couldn't make sure if it was well-designed.

Recently, 2 community users mentioned it and want this feature (https://github.com/pingcap/tiproxy/issues/609 and https://asktug.com/t/topic/1033941).  The current design can satisfy their needs, so it's time to expose it to users.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiproxy/tiproxy-load-balance.md [hunk 1]
```diff
@@ -5,23 +5,78 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+## 基于标签的负载均衡
+
+基于标签的负载均衡优先将连接路由到与 TiProxy 自身有相同标签的 TiDB server 上，以实现计算层的资源隔离。要启用基于标签的负载均衡，需要通过 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 指定用于匹配的标签名，TiProxy 通过该标签名查找 TiProxy 的 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 配置和 TiDB server 上的 [`labels`](/tidb-configuration-file.md#labels) 配置，并优先将连接路由到与自身的标签值相同的 TiDB server 上。
+
+例如，应用有交易和 BI 两类业务，为了避免相互影响，可以如下配置集群：
+
+- TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `app`，表示将按照标签名 `app` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
+- 配置 2 台 TiProxy 实例，配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。
+- 将 TiDB 实例分为 2 组，配置项 [`labels`](/tidb-configuration-file.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。 
+- 配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md) 以实现存储层的资源隔离。
```
Comment (joey-yez) [inline tiproxy/tiproxy-load-balance.md:35]: 这个不是必须，最好标注这一点是可选的，需要根据业务需要来判断是否进行配置。

File: tiproxy/tiproxy-load-balance.md [hunk 2]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+基于标签的负载均衡优先将连接路由到与 TiProxy 自身有相同标签的 TiDB server 上，以实现计算层的资源隔离。该策略默认不启用，仅当你的业务需要隔离计算层的资源时才需要启用。
+
+要启用基于标签的负载均衡，需要通过 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 指定用于匹配的标签名，TiProxy 通过该标签名查找 TiProxy 的 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 配置和 TiDB server 上的 [`labels`](/tidb-configuration-file.md#labels) 配置，并优先将连接路由到与自身的标签值相同的 TiDB server 上。
+
+例如，应用有交易和 BI 两类业务，为了避免相互影响，可以如下配置集群：
+
+- TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `app`，表示将按照标签名 `app` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
+- 配置 2 台 TiProxy 实例，配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。
+- 将 TiDB 实例分为 2 组，配置项 [`labels`](/tidb-configuration-file.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。 
+- 配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md) 以实现存储层的资源隔离。
```
Comment (joey-yez) [inline tiproxy/tiproxy-load-balance.md:37]: 示例中的这个步骤，配置 plancement rule，最好说明一下这个是可选的，否则可能让用户觉得很复杂，劝退潜在的试用。

Comment (joey-yez) [inline tiproxy/tiproxy-load-balance.md:37]:
```suggestion
- 若需要隔离存储资源，可以配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md) 以实现存储层的资源隔离。
```

**Review by joey-yez — APPROVED** [review]

File: tiproxy/tiproxy-configuration.md [hunk 3]
```diff
@@ -113,6 +113,12 @@ HTTP 网关的配置。
 
 TiProxy 负载均衡策略的配置。
 
+#### `label-name`
+
++ 默认值：`""`
++ 支持热加载：是
++ 指定[基于标签的负载均衡](/tiproxy/tiproxy-load-balance.md#基于标签的负载均衡)时使用的标签名，TiProxy 根据该标签名匹配 TiDB server 的标签值，并优先将请求路由到与自身有相同标签值的 TiDB server。配置该项后，TiProxy 需要配置 [`labels`](#labels)，TiDB server 需要配置 [`labels`](/tidb-configuration-file.md#labels)。空字符串代表不使用基于标签的负载均衡。有关详细信息，请参阅[基于标签的负载均衡](/tiproxy/tiproxy-load-balance.md#基于标签的负载均衡)。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:120]:
```suggestion
+ 指定用于[基于标签的负载均衡](/tiproxy/tiproxy-load-balance.md#基于标签的负载均衡)的标签名。TiProxy 根据该标签名匹配 TiDB server 的标签值，并优先将请求路由到与自身具有相同标签值的 TiDB server。
+ `label-name` 的默认值为空字符串，表示不使用基于标签的负载均衡。要启用该负载均衡策略，需要将此配置项设置为非空字符串，并配置 TiProxy 的 [`labels`](#labels) 和 TiDB 的 [`labels`](/tidb-configuration-file.md#labels) 配置项。有关详细信息，请参阅[基于标签的负载均衡](/tiproxy/tiproxy-load-balance.md#基于标签的负载均衡)。
```

File: tiproxy/tiproxy-load-balance.md [hunk 4]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
 
 TiProxy 默认启用所有策略，优先级从高到低依次为：
 
 1. 基于状态的负载均衡：当某个 TiDB server 正在关闭时，TiProxy 将连接从该 TiDB server 迁移到在线的 TiDB server。
-2. 基于健康度的负载均衡：当某个 TiDB server 的健康度异常时，TiProxy 将连接从该 TiDB server 迁移到健康度正常的 TiDB server。
-3. 基于内存的负载均衡：当某个 TiDB server 存在 Out of Memory (OOM) 风险时，TiProxy 将连接从该 TiDB server 迁移到内存使用量较低的 TiDB server。
-4. 基于 CPU 的负载均衡：当某个 TiDB server 的 CPU 使用率远高于其他 TiDB server 时，TiProxy 将连接从该 TiDB server 迁移到 CPU 使用率较低的 TiDB server。
-5. 基于地理位置的负载均衡：优先将请求路由到地理位置上距离 TiProxy 较近的 TiDB server。
-6. 基于连接数的负载均衡：当某个 TiDB server 的连接数远高于其他 TiDB server 时，TiProxy 将连接从该 TiDB server 迁移到连接数较少的 TiDB server。
+2. 基于标签的负载均衡：优先将请求路由到与 TiProxy 实例自身有相同标签的 TiDB server，以实现计算层的资源隔离。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:13]:
```suggestion
2. 基于标签的负载均衡：优先将请求路由到与 TiProxy 实例自身具有相同标签的 TiDB server，以实现计算层的资源隔离。
```

File: tiproxy/tiproxy-load-balance.md [hunk 5]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
 
 如需调整负载均衡策略的优先级，请参考[负载均衡策略配置](#负载均衡策略配置)。
 
 ## 基于状态的负载均衡
 
 TiProxy 定时通过 SQL 端口和状态端口检查 TiDB 是否已下线或正在关闭。
 
+## 基于标签的负载均衡
+
+基于标签的负载均衡优先将连接路由到与 TiProxy 自身有相同标签的 TiDB server 上，以实现计算层的资源隔离。该策略默认不启用，仅当你的业务需要隔离计算层的资源时才需要启用。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:28]:
```suggestion
基于标签的负载均衡优先将连接路由到与 TiProxy 自身具有相同标签的 TiDB server 上，从而实现计算层的资源隔离。该策略默认关闭，仅当你的业务需要隔离计算层的资源时才需要启用。
```

File: tiproxy/tiproxy-load-balance.md [hunk 6]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+
+要启用基于标签的负载均衡，需要通过 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 指定用于匹配的标签名，TiProxy 通过该标签名查找 TiProxy 的 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 配置和 TiDB server 上的 [`labels`](/tidb-configuration-file.md#labels) 配置，并优先将连接路由到与自身的标签值相同的 TiDB server 上。
+
+例如，应用有交易和 BI 两类业务，为了避免相互影响，可以如下配置集群：
+
+- TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `app`，表示将按照标签名 `app` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:34]:
```suggestion
- 在 TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `"app"`，表示将按照标签名 `"app"` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
```

File: tiproxy/tiproxy-load-balance.md [hunk 7]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+- 配置 2 台 TiProxy 实例，配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:35]:
```suggestion
- 配置 2 台 TiProxy 实例，分别为配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 加上 `"app"="Order"` 和 `"app"="BI"`。
```

File: tiproxy/tiproxy-load-balance.md [hunk 8]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+- 将 TiDB 实例分为 2 组，配置项 [`labels`](/tidb-configuration-file.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:36]:
```suggestion
- 将 TiDB 实例分为 2 组，分别为配置项 [`labels`](/tidb-configuration-file.md#labels) 加上 `"app"="Order"` 和 `"app"="BI"`。 
```

File: tiproxy/tiproxy-load-balance.md [hunk 9]
```diff
@@ -5,23 +5,80 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+基于标签的负载均衡优先将连接路由到与 TiProxy 自身有相同标签的 TiDB server 上，以实现计算层的资源隔离。该策略默认不启用，仅当你的业务需要隔离计算层的资源时才需要启用。
+
+要启用基于标签的负载均衡，需要通过 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 指定用于匹配的标签名，TiProxy 通过该标签名查找 TiProxy 的 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 配置和 TiDB server 上的 [`labels`](/tidb-configuration-file.md#labels) 配置，并优先将连接路由到与自身的标签值相同的 TiDB server 上。
+
+例如，应用有交易和 BI 两类业务，为了避免相互影响，可以如下配置集群：
+
+- TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `app`，表示将按照标签名 `app` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
+- 配置 2 台 TiProxy 实例，配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。
+- 将 TiDB 实例分为 2 组，配置项 [`labels`](/tidb-configuration-file.md#labels) 分别加上 `"app"="Order"` 和 `"app"="BI"`。 
+- 如果需要同时隔离存储层的资源，可配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md) 。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:37]:
```suggestion
- 如果需要同时隔离存储层的资源，可配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md)。
```

File: tiproxy/tiproxy-grafana.md [hunk 10]
```diff
@@ -59,6 +59,7 @@ TiProxy 有四个面板组。这些面板上的指标表示 TiProxy 的当前状
 - Session Migration Duration：会话迁移的平均、P95、P99 时长。
 - Session Migration Reasons：每分钟发生的会话迁移数，以及迁移的原因。原因包括：
     - `status`：TiProxy 进行了[基于状态的负载均衡](/tiproxy/tiproxy-load-balance.md#基于状态的负载均衡)
+    - `label`：TiProxy 进行了[基于标签的负载均衡](/tiproxy/tiproxy-load-balance.md#基于标签的负载均衡)
```
Comment (hfxsd) [inline tiproxy/tiproxy-grafana.md:62]: 应该是 `label-name`？

File: tiproxy/tiproxy-load-balance.md [hunk 11]
```diff
@@ -5,23 +5,86 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
 
+## 基于标签的负载均衡
+
+基于标签的负载均衡优先将连接路由到与 TiProxy 自身具有相同标签的 TiDB server 上，从而实现计算层的资源隔离。该策略默认关闭，仅当你的业务需要隔离计算层的资源时才需要启用。
+
+要启用基于标签的负载均衡，你需要：
+
+- 通过 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 指定用于匹配的标签名
+- 配置 TiProxy 配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels)
+- 配置和 TiDB server 配置项 [`labels`](/tidb-configuration-file.md#labels)
```
Comment (hfxsd) [inline tiproxy/tiproxy-load-balance.md:34]:
```suggestion
- 配置 TiDB server 配置项 [`labels`](/tidb-configuration-file.md#labels)
```

File: tiproxy/tiproxy-load-balance.md [hunk 12]
```diff
@@ -5,23 +5,86 @@ summary: 介绍 TiProxy 的负载均衡策略及其适用场景。
...
+
+配置完成后，TiProxy 会根据 `balance.label-name` 指定的标签名查找相应的配置，并将连接路由到标签值相同的 TiDB server。
+
+例如，若应用包含交易和 BI 两类业务，为了避免相互影响，可以按照如下方式配置集群：
+
+- 在 TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `"app"`，表示将按照标签名 `"app"` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
+- 配置 2 台 TiProxy 实例，分别为配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 加上 `"app"="Order"` 和 `"app"="BI"`。
+- 将 TiDB 实例分为 2 组，分别为配置项 [`labels`](/tidb-configuration-file.md#labels) 加上 `"app"="Order"` 和 `"app"="BI"`。 
+- 如果需要同时隔离存储层的资源，可配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md)。
+- 交易和 BI 业务的客户端分别连接到 2 台 TiProxy 的地址。
```
Comment (hfxsd) [inline tiproxy/tiproxy-load-balance.md:44]:
```suggestion
1. 在 TiProxy 上配置 [`balance.label-name`](/tiproxy/tiproxy-configuration.md#label-name) 为 `"app"`，表示将按照标签名 `"app"` 匹配 TiDB server，并将连接路由到相同标签值的 TiDB server 上。
2. 配置 2 台 TiProxy 实例，分别为配置项 [`labels`](/tiproxy/tiproxy-configuration.md#labels) 加上 `"app"="Order"` 和 `"app"="BI"`。
3. 将 TiDB 实例分为 2 组，分别为配置项 [`labels`](/tidb-configuration-file.md#labels) 加上 `"app"="Order"` 和 `"app"="BI"`。 
4. 如果需要同时隔离存储层的资源，可配置 [Placement Rules](/configure-placement-rules.md) 或[资源管控](/tidb-resource-control.md)。
5. 交易和 BI 业务的客户端分别连接到 2 台 TiProxy 的地址。
```

**Review by hfxsd — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

File: tiproxy/tiproxy-grafana.md (hunk 10 above)
Comment (djshow832) [reply tiproxy/tiproxy-grafana.md:62]: 不，就是 `label`。

**Pull request merged**